### PR TITLE
Add a generate command to create and recovery keys.

### PR DESCRIPTION
**Pull request by jplevyak**
This is bip39 and keysmith compatible.

### CODE REVIEW

**Comment by jplevyak** [comment]
https://github.com/dfinity/quill/issues/60

**Review by chmllr — COMMENTED** [review]
Thanks so much John! Please increase the version in Cargo.toml too.

File: src/commands/generate.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                0,
+                BigUint::from(1u32),
+                Box::new(BitString(0, public_key.len() * 8, public_key)),
+            ),
+        ],
+    );
+    to_der(&data).expect("Cannot encode secret key.")
+}
+
+/// Prints the account and the principal ids.
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply src/commands/generate.rs]: done

File: src/commands/generate.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+pub fn exec(opts: GenerateOpts) -> AnyhowResult {
+    let seed_file = match opts.seed_file {
+        Some(f) => f,
+        None => String::from("seed.txt"),
+    };
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply src/commands/generate.rs]: done

File: src/commands/generate.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    if Path::exists(Path::new(&seed_file)) && !opts.overwrite_seed_file {
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply src/commands/generate.rs]: done

File: src/commands/generate.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                return Err(anyhow!("Words must be 12 or 24."));
+            }
+            if words > 12 {
+                32
+            } else {
+                16
+            }
+        }
+        None => 16,
+    };
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply src/commands/generate.rs]: done

**Review by chmllr — APPROVED** [review]

**Pull request merged**